src/os/bluestore/BlueStore.cc

```diff
@@ -8605,6 +8605,16 @@ int BlueStore::dump_bluefs_sizes(ostream& out)
   return r;
 }
 
+void BlueStore::trim_free_space()
+{
+  auto iterated_allocation = [&](size_t off, size_t len) {
+    ceph_assert(len > 0);
+    bdev->_discard(off, len);
```

Comment:
Not completely sure but perhaps we should filter out rotational devices. What do you think?

Reply:
I think we can achieve this by adding the support_discard check in the discard function